applications/FluidDynamicsHydraulicsApplication/python_scripts/apply_hydraulic_inlet_process.py

```diff
@@ -0,0 +1,181 @@
+from math import sqrt
+import KratosMultiphysics
+import KratosMultiphysics.FluidDynamicsHydraulicsApplication as KratosFluidHydraulics
+from KratosMultiphysics.read_csv_table_utility import ReadCsvTableUtility
+import KratosMultiphysics.kratos_utilities as KratosUtils
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ApplyHydraulicInletProcess(Model, settings["Parameters"])
+
+
+class ApplyHydraulicInletProcess(KratosMultiphysics.Process):
+    """
+    Class to compute hydraulic inlet profile
+    """
+    def __init__(self, Model, settings):
+        KratosMultiphysics.Process.__init__(self)
+
+        # Set default settings
+        # Note the trick to allow scalar, function and table values of the inlet discharge
+        default_settings = self.GetDefaultParameters()
+        if settings.Has("value"):
+            if settings["value"].IsString():
+                default_settings["value"].SetString("0.0")
+            elif settings["value"].IsDouble():
+                default_settings["value"].SetDouble(0.0)
+        else:
+            raise Exception("'value' not found. It needs to be user-provided.")
+
+        self.interval = KratosMultiphysics.IntervalUtility(settings)
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        # Get the inlet model part , the inlet free surface (water depth) variable name and user tolerances.
+        self.inlet_model_part = Model[settings["inlet_model_part_name"].GetString()]
+        variable_name =settings["water_depth_variable"].GetString()
+        self.water_depth_variable = KratosMultiphysics.KratosGlobals.GetVariable(variable_name)
+        self.bisection_tolerance = settings["critical_depth_tolerance"].GetDouble()
```

Comment:
```suggestion
        self.critical_depth_tolerance = settings["critical_depth_tolerance"].GetDouble() # Bisection tolerance
```
Very minor, but it is more handy to keep the input name for the corresponding member variable, specially in need of eventual debugging.